src/Account.sol

```diff
@@ -635,32 +645,56 @@ contract Account is IAccount, Auth, OpsReady {
             MARGIN_ASSET.transfer(SETTINGS.TREASURY(), fee);
         }
 
-        /// @custom:todo add event emission for order flow fee imposed
+        EVENTS.emitOrderFlowFeeImposed({amount: fee});
+    }
+
+    /// @notice impose an order flow fee on the account
+    /// @param _market: address of market
+    /// @param _sizeDelta: size delta of order
+    /// @dev will attempt to deduct fee from account's idle margin first
+    /// @dev if fee exceeds idle margin, fee will be deducted from market's margin
+    /// if possible. If not possible, the transaction will revert.
+    /// @dev if fee is deducted from market's margin, then the following order
+    /// may be rejected if the account has insufficient available market margin
+    function _imposeOrderFlowFee(address _market, int256 _sizeDelta) internal {
+        _imposeOrderFlowFee(_market, _sizeDelta, 0);
     }
 
     /// @notice calculate order flow fee for a given market and size delta
     /// @param _market: address of market
     /// @param _sizeDelta: size delta of order
+    /// @param _desiredFillPrice desiredFillPrice in case of a delayed Order
     /// @return fee: order flow fee to impose
-    function _calculateOrderFlowFee(address _market, int256 _sizeDelta)
-        internal
-        view
-        returns (uint256)
-    {
+    /// @dev _desiredFillPrice is used to calculate orderFlowFee for Delayed Orders
+    function _calculateOrderFlowFee(
+        address _market,
+        int256 _sizeDelta,
+        uint256 _desiredFillPrice
+    ) internal view returns (uint256) {
         // fetch order flow fee from settings
         uint256 orderFlowFee = SETTINGS.orderFlowFee();
 
         // fetch current sUSD exchange rate for market
         IPerpsV2MarketConsolidated market = IPerpsV2MarketConsolidated(_market);
         (uint256 price,) = _sUSDRate(market);
 
+        uint256 usedPrice = (_desiredFillPrice != 0) ? _desiredFillPrice : price;
```

Comment:
if we use `_desiredFillPrice` there is no need to query `_sUSDRate`, right? lets move this line up and only call `_sUSDRate` if necessary

Reply:
something like this
```
uint price;
if (_desiredFillPrice != 0) {
    price = _desiredFillPrice;
} else {
    (price,) = _sUSDRate(IPerpsV2MarketConsolidated(_market));
}
```

Reply:
Updated